core/state-machine/src/changes_trie/storage.rs

```diff
@@ -37,8 +37,8 @@ pub struct InMemoryStorage<H: Hasher, Number: BlockNumber> {
 }
 
 /// Adapter for using changes trie storage as a TrieBackendEssence' storage.
-pub struct TrieBackendAdapter<'a, H: Hasher, Number: BlockNumber, S: 'a + Storage<H, Number>> {
-	storage: &'a S,
+pub struct TrieBackendAdapter<'a, H: Hasher, Number: BlockNumber> {
+	storage: &'a dyn Storage<H, Number>,
```

Comment:
iirc - this is somehow related to new `struct State` from [here](https://github.com/paritytech/substrate/pull/3201/files#diff-4a0480641c07192d5cf9317a41286b93R106), that'll be used instead of simply passing `&'a S`. Sorry - can't just remember it now whether it is the obliged change, or it was just me tired of making everything generic over `Storage`. I could try to revert and find the original reason, if you insist.

Reply:
No that's all fine I prefer impl, but it is a personal thing and not something that shall be enforce :)